pkg/remotedialer/wsconn.go

```diff
@@ -24,6 +24,7 @@ func newWSConn(conn *websocket.Conn) *wsConn {
 func (w *wsConn) WriteMessage(messageType int, data []byte) error {
 	w.Lock()
 	defer w.Unlock()
+	w.conn.SetWriteDeadline(time.Now().Add(PingWaitDuration))
```

Comment:
We also write a message inside the same lock at line 39. Do you think the deadline should be set there as well?

Reply:
The last argument of WriteControl is the deadline.  I don't know why the API is like that.